mantle/kola/harness.go

```diff
@@ -672,15 +694,20 @@ func runProvidedTests(testsBank map[string]*register.Test, patterns []string, mu
 
 func getRerunnable(tests []*harness.H) []string {
 	var testsToRerun []string
+	expPrefix := regexp.MustCompile(`^non-exclusive-test-bucket-[0-9]/`)
+	exp := regexp.MustCompile(`^non-exclusive-test-bucket-[0-9]$`)
```

Comment:
we use these compiled regex's in two different places and the logic isn't very easy to follow. I think it could be worth us creating a function that does this for us and gives us meaningful information back without us having to think about it.

i.e. given this test name I want the following information:

- Is this the main nonexclusive test wrapper or a real test?
- What is the test name

In that separate function we could comment it with all the meaningful details about the naming convention for the nonexclusive tests and what is a wrapper/not a wrapper, etc.. I think calling the function would help people follow the logic easier and if they're interested in the details they can go look at the function (that is nicely commented with the details). WDYT?

Reply:
I agree, I think this is a good idea.